copy.c

```diff
@@ -1,11 +1,46 @@
 #include <stdio.h>
 #include <windows.h>
+#include <fcntl.h>
+
+typedef struct node_tag {
+	char buffer[1024];
+	int capacity;
+	struct node_tag *next;
+} node;
+
+int wmain() {
+	_setmode(_fileno(stdin), _O_U8TEXT);
+
+	node* bufferList = malloc(sizeof(node));
+	bufferList->capacity = 0;
+	node* head = bufferList;
+
+	DWORD cbSize = 0;
+	DWORD count;
+	while ((count = fread(head->buffer, 1, sizeof(head->buffer), stdin)) != 0 && !ferror(stdin)) {
+		cbSize += count;
+		head->capacity = count;
+		head->next = malloc(sizeof(node));
+		head->next->capacity = 0;
+	}
+
+	if (ferror(stdin)) {
+		return 1;
+	}
 
-int wmain(int argc, wchar_t **argv) {
-	DWORD cbSize = (wcslen(argv[1]) + 1) * 2;
 	HANDLE hgMem = GlobalAlloc(GMEM_MOVEABLE, cbSize);
 	LPBYTE lpData = GlobalLock(hgMem);
-	CopyMemory(lpData, argv[1], cbSize);
+
+	cbSize = 0;
+	head = bufferList;
+	while (head->capacity != 0) {
+		CopyMemory(lpData + cbSize, head->buffer, head->capacity);
+		cbSize += head->capacity;
+		node* prev = head;
```

Comment:
Super nitpickey - `node *prev` instead of `node* prev`. [This answer explains why](http://stackoverflow.com/a/8685467/510036), but I'm in the second camp. :)